ngReact.js

```diff
@@ -82,6 +113,31 @@
     }, {});
   }
 
+  /**
+   *
+   * @param watchDepth (value of HTML watch-depth attribute)
+   * @param scope (angular scope)
+   *
+   * Uses the watchDepth attribute to determine how to watch props on scope.
+   * If watchDepth attribute is NOT reference or collection, watchDepth defaults to deep watching by value
+   */
+  function watchProps (watchDepth, scope){
+    var args = Array.prototype.slice.call(arguments, 2);
+    var watchFn;
+
+    //default watchDepth to value if not reference or collection
+    if (isWatchCollectionAvailable && watchDepth === 'collection') {
```

Comment:
Would a simpler `if ( angular.isFunction( '$watchCollection' && .. ` not be sufficient. The nice thing about this approach is that the detection doesn't happen based on an exact version.